ports/geckolib/gecko_bindings/sugar/ns_style_coord.rs

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use bindings::{Gecko_ResetStyleCoord, Gecko_SetStyleCoordCalcValue, Gecko_AddRefCalc};
+use std::mem::transmute;
+use structs::{nsStyleCoord_CalcValue, nsStyleCoord_Calc, nsStyleUnit, nsStyleUnion, nsStyleCoord};
+
+// Functions here are unsafe because it is possible to use the wrong nsStyleUnit
+// FIXME we should be pairing up nsStyleUnion and nsStyleUnit somehow
+// nsStyleCoord is one way to do it, but there are other structs using pairs
+// of union and unit too
```

Comment:
I was imagining that we'd create a some temporary tuple struct that isn't actually nsStyleCoord, but if we can make that nsStyleCoord itself with similar properties, that's even better.